### ISSUE TITLE
Inconsistencies in Model Registration and Unregistration Functions in `models.py`

### ISSUE THREAD

**Issue by Geogouz** [rucio/rucio#6677]
I noticed couple of inconsistencies in the `models.py` related to how models are registered and unregistered.

For example:
1. **Duplicate Entries**: The model `ReplicationRule` is listed twice in the `register_models` function (even though SQLAlchemy shouldn't have a problem handling this).

https://github.com/rucio/rucio/blob/339744866e9550f0f068f8e3c9574a95ccaa7d59/lib/rucio/db/sqla/models.py#L1740-L1743

2. **Missing Model in Unregistration Function**: The `VirtualPlacements` model is included in the `register_models` function but is absent from the `unregister_models` function.

3. **Missing Models from both Functions**: The models `BadPFNs`, `TransferLimit`, `RSETransferLimit`, `DatasetLock`, `Distance`, `TransferStats` are not considered at all.

Are (any of) these intensional?

**Comment by erlingstaff** [comment]
The current implementation is very confusing and not ideal. Even though there are models missing in register and unregister, as you noted, they all get created and deleted correctly (the commands in psql screenshot are run before/after running `./tools/destroy_database.py`, which calls `unregister_models`). 

<img width="800" alt="Pasted Graphic 1" src="https://github.com/rucio/rucio/assets/38101463/b39fc428-9da7-4dde-95fc-ccba2255ce2b">
 
 When unregister or register_models is run now, it loops over all the listed models and calls `model.metadata.drop_all(engine)`, where metadata is a db schema, from which all tables (models) are dropped. However, they all inherit the same metadata and schema from the `BASE` class.
 
 TL;DR:
 Currently `register_models` tries to register the same schema many times (one time for each model listed), likewise with `unregister_models`. The schema contains all the models. Simplified down to a single call `BASE.metadata.drop_all(engine)` /  `BASE.metadata.create_all(engine)` to register or unregister the schema. #6712
